include/superior_mysqlpp/low_level/dbdriver.hpp

```diff
@@ -319,6 +319,28 @@ namespace SuperiorMySqlpp { namespace LowLevel
             return escapeStringNoConnection(original.c_str(), original.length());
         }
 
+        // XXX: makeHexString functions can be implemented by one function using constexpr-if from c++17
+        /**
+         * Converts string to HEX string suitable for using in SQL statements.
+         * Useful for binary data for instance.
+         * @see https://dev.mysql.com/doc/refman/5.7/en/mysql-hex-string.html
+         *
+         * @tparam Type convertable to char[]
+         * @param from Original string.
+         * @tparam Size type.
+         * @param length of original string
+         * @return HEX string.
+         */
+        template <typename T>
+        std::string makeHexString(T *form, std::size_t formlen)
+        {
+
+            auto buffer = std::make_unique<char[]>(formlen * 2 + 1);
```

Comment:
I would rather create std::string, resize it, use it as a buffer and then return it directly to the caller (there will be only one allocation instead of two)

Reply:
You are right, I actually tried it, but it works only when I used this ugly hack with -1.
```C++
    std::string result(formlen * 2 + 1, '\0');

    auto length = mysql_hex_string(const_cast<char *>(result.data() - 1), form, formlen);
    result.resize(length);

    return result;

```
If I don't use -1 first character of string remains the same.
I didn't want to use this version because it's odd.

Reply:
Did you tried to use `&*result.begin()`? And then, you don't need that extra null byte, it is there by default.

```c++
std::string result;
result.resize(formlen * 2);
result.resize(mysql_hex_string(&*result.begin(), form, formlen));
return result;
```